test/object_store/conftest.py

```diff
@@ -44,12 +45,28 @@ def ssl(request):
     yield request.config.getoption('--ssl')
 
 
+@contextmanager
+def _maybe_remove_tempdir_on_complete(request, tempdir):
```

Comment:
The size of hacks around tempdir fixture is becoming ... rather large. Maybe drop the fixture and maintain tempdir by hand, like the cql-pytest's run_with_temporary_dir prototype did?

Reply:
you mean using global registry for tracking the cleanup? i think it is a stepback. i can, however drop the `_maybe_remove_tempdir_on_complete` and move it into `test_tempdir()`, despite that they serve the different purposes.

Reply:
meh... global registry is, probably, not much better. Let's try with this one